### ISSUE TITLE
[all] image is hosted on dockerhub

### ISSUE THREAD

**Issue by rptaylor** [kubernetes/cloud-provider-openstack#1753]
/kind feature

**What happened**:
Deployment fails on large clusters because of dockerhub rate limiting.
https://github.com/kubernetes/cloud-provider-openstack/blob/master/charts/cinder-csi-plugin/values.yaml#L49
```
rpc error: code = Unknown desc = Error response from daemon: toomanyrequests:
You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
```

**What you expected to happen**:
The cinder-csi-plugin image should be hosted on a container registry that is suitable for production use, so users can deploy it without being blocked by Dockerhub's restrictive rate limiting policy. Dockerhub should only be used for small-scale testing and development.
Besides, each other image (csi-attacher, csi-provisioner, csi-snapshotter, csi-resizer, livenessprobe, and csi-node-driver-registrar) is hosted on k8s.gcr.io. Can the cinder image be hosted there too for consistency, as well as providing suitable hosting capacity for production use?

**How to reproduce it**:
Deploy the cinder CSI helm chart on a large cluster.

**Anything else we need to know?**:

**Environment**:
- openstack-cloud-controller-manager(or other related binary) version:
- OpenStack version:
- Others:

**Comment by jichenjc** [comment]
@lingxiankong @ramineni I don't remember clearly but seems we have this discussion somewhere before
I am also suffering from the pull limit from docker hub so guess this is not uncommon

seems other repo such as 
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/Makefile#L73
can do this for openstack ,so maybe we can follow as well?

**Comment by ramineni** [comment]
@jichenjc +1 to host on k8s.gcr.io . we need to check the process, credentials required for the same.

**Comment by lingxiankong** [comment]
Agree we could upload the images to k8s.gcr.io, but hosting in dockerhub doesn't mean it is not suitable for production use case as the issue said.

> Dockerhub should only be used for small-scale testing and development.

This is totally wrong.

The common solution of container image in large scale cluster is either hosting the container images in private/enterprise registry service or providing dockerhub enterprise account to avoid the rate limit issue.

Personally, I don't think using dockerhub is an issue especially for the upstream project like this, but I'm open to other options.

**Comment by rptaylor** [comment]
> Agree we could upload the images to k8s.gcr.io

@lingxiankong  Yes that would be great, could you do that please?

Anyone using a ~> medium size cluster (~ 100 nodes) will run into this issue with images hosted on Dockerhub. Projects have been moving away from hosting containers on Dockerhub since the rate limiting was introduced as there are free viable alternatives that do not block users from deploying moderate numbers of containers.

> The common solution of container image in large scale cluster is either hosting the container images in private/enterprise registry service

In other words, not using Dockerhub. Yes, but then we would have to copy the image each time it is updated, and modify the helm chart for every application we deploy, and so would every other user running cinder CSI at a modest scale. Why not fix the problem for everybody, and improve consistency in the cloud-provider-openstack project by hosting the cinder-csi plugin image in the same place as the csi-attacher, csi-provisioner, csi-snapshotter, csi-resizer, livenessprobe, and csi-node-driver-registrar images?

>  or providing dockerhub enterprise account 

I don't think that requiring users to sign up for an account on a 3rd party service, nor paying to download free and open source software, is an appropriate solution, especially for organizations that are (like us) working in non-profit scientific research, not the commercial sector.

**Comment by lingxiankong** [comment]
> > Agree we could upload the images to k8s.gcr.io
> 
> @lingxiankong Yes that would be great, could you do that please?

The upstream procedure for feature request is we (the repo maintainer team) discuss and decide if a feature request is accepted or not, so we will close the issue based on the discussion. For this one, I can reopen that, and we will see if there will be some contributor pick up, and you are also welcome to offer the help.

> 
> Anyone using at least a ~ medium size cluster (~ 100 nodes) will run into this issue with images hosted on Dockerhub. Projects have been moving away from hosting containers on Dockerhub since the rate limiting was introduced as there are free viable alternatives that do not block users from deploying moderate numbers of containers.
> 
> > The common solution of container image in large scale cluster is either hosting the container images in private/enterprise registry service
> 
> In other words, not using Dockerhub. Yes, but then we would have to copy the image each time it is updated, and modify the helm chart for every application we deploy, and so would every other user running cinder CSI at a modest scale. Why not fix the problem for everybody, and improve consistency in the cloud-provider-openstack project by hosting the cinder-csi plugin image in the same place as the csi-attacher, csi-provisioner, csi-snapshotter, csi-resizer, livenessprobe, and csi-node-driver-registrar images?

Again, using Dockerhub is not a problem from the open source community point of view, I didn't see there is such hard requirement defined in kubernetes community (please correct me if I'm wrong) .The project just happened to use Dockerhub in the beginning and now it's not suitable for some organizations who don't have their own image registry service solution. In most of the companies or organizations I've known, it's not uncommon to set up a sync mechanism to pull new images when new versions are released and updating the image tag in either helm chart or kustomize config.

That being said, I do hear you and I support if someone could jump in and help to set up the CI to upload images to k8s.gcr.io.

**Comment by lingxiankong** [comment]
For those who are interested:

https://github.com/kubernetes/test-infra/blob/master/config/jobs/image-pushing/README.md

**Comment by ramineni** [comment]
@rptaylor As @lingxiankong mentioned, we are happy to support it if someone willing to take this up.

This issue is duplicate of https://github.com/kubernetes/cloud-provider-openstack/issues/1540 
So closing the issue. we could track the progress as part of the above issue.

**Comment by rptaylor** [comment]
Okay, thanks for your consideration @lingxiankong . I agree it is not a firm requirement, rather an attempt to improve useability and convenience for some users. Will follow up in the other issue.
